src/main/java/com/diffplug/gradle/pde/PdeProductBuildConfig.java

```diff
@@ -89,8 +90,11 @@ void setup(File destinationDir, PdeBuildProperties props, List<SwtPlatform> plat
 
 		// now create the sanitized product file
 		File productFile = productPluginDir.toPath().resolve(productFileWithinPlugin).toFile();
+		productFileLines = ProductFileUtil.readLines(productFile);
+		ProductFileUtil.extractProperties(productFileLines).forEach((key, value) -> props.setProp(key.toString(), value.toString()));
+
```

Comment:
After making the `Map<String, String>` change, this could be `.forEach(props::setProp)`